lib/stats/stats-views.c

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2017 Balabit
+ * Copyright (c) 2017 Noemi Vanyi
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+
+#include "stats-views.h"
+
+static void
+_calculate_written_messages(GList *counters, StatsCounterItem **result)
+{
+  StatsCounterItem *processed = NULL, *dropped = NULL, *queued = NULL;
+  gssize written = 0;
+
+  for (GList *c = counters; c; c = c->next)
```

Comment:
Does `counters` contain only one instance of the selected types?
What if one of the types is not presented in the `counters` list?